src/video/movies.cpp

```diff
@@ -178,8 +184,92 @@ uint32_t ffmpeg_prepare_movie(char *name, bool with_audio)
 	movie_fps = av_q2d(av_guess_frame_rate(format_ctx, format_ctx->streams[videostream], NULL));
 	movie_duration = (double)format_ctx->duration / (double)AV_TIME_BASE;
 	movie_frames = (uint32_t)::round(movie_fps * movie_duration);
-	movie_jpeg_range = codec_ctx->color_range == AVCOL_RANGE_JPEG;
-
+	fullrange_input = (codec_ctx->color_range == AVCOL_RANGE_JPEG);
+    
+    // some pixel formats are inherently full-range
+    // so we should treat them as such, even if the color range metadata is missing
+    switch (codec_ctx->pix_fmt){
+        case AV_PIX_FMT_YUVJ420P:
+        case AV_PIX_FMT_YUVJ411P:
+        case AV_PIX_FMT_YUVJ422P:
+        case AV_PIX_FMT_YUVJ444P:
+        case AV_PIX_FMT_YUVJ440P:
+            fullrange_input = true;
+            yuvjfixneeded = true;
+            break;
+        case AV_PIX_FMT_GRAY8:
+        case AV_PIX_FMT_YA8:
+        case AV_PIX_FMT_GRAY16LE:
+        case AV_PIX_FMT_GRAY16BE:
+        case AV_PIX_FMT_YA16BE:
+        case AV_PIX_FMT_YA16LE:
+            fullrange_input = true;
+            yuvjfixneeded = false;
+            break;
+        default:
+            yuvjfixneeded = false;
+    }
+        
+    // will we need to convert the colorspace?
+    switch(codec_ctx->colorspace){
+        // these are all the same (bt601)
+        case AVCOL_SPC_UNSPECIFIED: // ffmpeg guesses and treats this as bt601 
+        case AVCOL_SPC_RESERVED: // ffmpeg guesses and treats this as bt601 
+        case AVCOL_SPC_BT470BG:
+        case AVCOL_SPC_SMPTE170M:
+            okcolorspace = true;
+            break;
+        default:
+            okcolorspace = false;
+    }
+    
+    // what gamma should we use?
+    switch(codec_ctx->color_trc){
+        case AVCOL_TRC_IEC61966_2_1: //srgb
+        case AVCOL_TRC_UNSPECIFIED: //assume srgb
+        case AVCOL_TRC_RESERVED: //assume srgb
+        case AVCOL_TRC_RESERVED0: //assume srgb
+        case AVCOL_TRC_GAMMA22: //another alias
+            use170Mgamma = false;
+            break;
+        case AVCOL_TRC_SMPTE170M:
+        case AVCOL_TRC_BT709: // same as SMPTE170M
+        case AVCOL_TRC_IEC61966_2_4: // same as 709
+        case AVCOL_TRC_BT1361_ECG: // not quite sure on this one, but think it's the same curve
+            use170Mgamma = true;
+            if (trace_movies)
+            {
+                ffnx_trace("prepare_movie: SMPTE170M transfer function detected\n");
+            }
```

Comment:
Please compact this if to a one-liner. Thank you

Reply:
Done